dev/reference/tidb-binlog/faq.md

```diff
@@ -1,13 +1,139 @@
+
+- Add pump instances.
+
+## What can I do if Drainer synchronization is interrupted?
+
+Execute the following command to check whether the status of Pump is normal and whether all the Pump instances that are not in the `offline` state are running.
+
+{{< copyable "shell-regular" >}}
+
+``` bash
+binlogctl -cmd pumps
+```
+
+Then, check whether the Drainer monitor or log outputs corresponding errors. If so, resolve them accordingly.
+
+## What can I do if Drainer is slow to synchronizes the downstream TiDB/MySQL?
+
+Check the following monitoring items:
+
+- For the **Drainer Event** monitoring metric, check the speed of Drainer synchronizing `INSERT`, `UPDATE` and `DELETE` transactions to the downstream per second.
+
+- For the **SQL Query Time** monitoring metric, check the time Drainer takes to execute SQL statements in the downstream.
+
+Possible causes and solutions for slow synchronization:
+
+- If the synchronized database contains a table without a primary key or unique index, add a primary key to the table.
+
+- If the latency between Drainer and the downstream is high, increase the value of the `worker-count` parameter of Drainer. For cross-datacenter synchronization, it is recommended to deploy Drainer in the downstream.
+
+- If the load in the downstream is not high, increase the value of the `worker-count` parameter of Drainer.
+
+## What can I do if a Pump instance crashes?
+
+If a Pump instance crashes, Drainer cannot synchronize data to the downstream because it cannot obtain the data of this instance. If this Pump instance can return to normal use, Drainer resumes synchronization; if not, perform the following steps:
+
+1. Use [binlogctl to change the state of this Pump instance to `offline`](/dev/reference/tidb-binlog/maintain.md) to discard the data of this Pump instance.
+
+2. Because Drainer cannot obtain the data of this pump instance, the data in the downstream and upstream is inconsistent. In this situation, you need to perform full synchronization and incremental synchronization again. The steps are as follows:
+
+    i. Stop the Drainer.
+
+    ii. Perform a full backup in the upstream.
+
+    iii. Clear the data in the downstream including the `tidb_binlog.checkpoint` table.
+
+    iv. Synchronize the full backup data to the downstream.
+
+    v. Deploy Drainer and use `initialCommitTs` (set `initialCommitTs` as the snapshot timestamp of the full backup) as the start point of initial synchronization.
+
+## What is checkpoint?
+
+Checkpoint records the `commit-ts` that Drainer synchronizes to the downstream. When Drainer restarts, it reads the checkpoint and then synchronizes data to the downstream starting from the corresponding `commit-ts`. The `["write save point"] [ts=411222863322546177]` Drainer log means saving the checkpoint with the corresponding timestamp.
+
+Checkpoint is saved in different files for different types of downstream:
```

Comment:
s/different files/different way  ?

may save in file or MySQL/TiDB  not different files ?